include/config.h

```diff
@@ -48,6 +51,17 @@ namespace detail {
 		std::optional<bool> m_enable_device_profiling;
 		size_t m_graph_print_max_verts = 200;
 		int m_dry_run_nodes = 0;
+
+		// required for test
+		std::vector<size_t> m_devices_list;
```

Comment:
This doesn't just store devices, it's a platform followed by a list of devices. But I don't see why this is required anyway, can't you just check that `m_device_cfg` is set correctly?